src/main/java/net/backupcup/mcde/screen/handler/GildingFoundryScreenHandler.java

```diff
@@ -144,4 +173,62 @@ private void addPlayerHotbar(PlayerInventory playerInventory) {
             this.addSlot(new Slot(playerInventory, i, 10 + i * 18, 142));
         }
     }
+
+    public List<Identifier> getCandidatesForGidling() {
+        return EnchantmentUtils.getEnchantmentsForItem(inventory.getStack(0)).collect(Collectors.toList());
+    }
+
+    public Optional<Identifier> generateEnchantment(PlayerEntity player) {
+        return EnchantmentUtils.generateEnchantment(
+            inventory.getStack(0),
+            context.get((world, pos) -> world.getServer().getPlayerManager().getPlayer(player.getUuid())),
+            getCandidatesForGidling()
```

Comment:
As this is referring to the process of gilding, this *is* an appropriate choice of words to use `...Gilding`